drivers/SmartThings/zigbee-lock/src/init.lua

```diff
@@ -291,6 +293,41 @@ local name_slot = function(driver, device, command)
   end
 end
 
+local migrate = function(driver, device, command)
+  local lock_users = {}
+  local lock_credentials = {}
+  local lock_codes = lock_utils.get_lock_codes(device)
+  local ordered_codes = {}
+
+  for code in pairs(lock_codes) do
+    table.insert(ordered_codes, code)
+  end
+
+  table.sort(ordered_codes)
+  for index, code_slot in ipairs(ordered_codes) do
+    table.insert(lock_users, {userIndex = index, userType = "guest", userName = lock_codes[code_slot]})
```

Comment:
Zigbee supports other user types, and IIRC some locks like Yale do use them. It doesn't look like `lockCodes` kept track of that information, but could we try to get the type with the `Get User Type` command?